x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_event_type_signal.ts

```diff
@@ -13,3 +13,16 @@ export const buildEventTypeSignal = (doc: BaseSignalHit): object => {
     return { kind: 'signal' };
   }
 };
+
+/**
+ * Given a document this will return true if that document is a signal
+ * document. We can't guarantee the code will call this function with a document
+ * before adding the _source.event.kind = "signal" from "buildEventTypeSignal"
+ * so we do basic testing to ensure that if the object has the fields of:
+ * "signal.rule.id" then it will be one of our signals rather than a customer
+ * overwritten signal.
+ * @param doc The document which might be a signal or it might be a regular log
+ */
+export const isEventTypeSignal = (doc: BaseSignalHit): boolean => {
```

Comment:
Just making a note here: in the future we could re-work this function as a type guard for the event doc.